testpackage/small_test_definitions.sh

```diff
@@ -17,139 +17,230 @@ run_dir="run"
+#######
+# GLOBAL 2D TESTS (20..21)
+#######
+
+#Very small ecliptic magnetosphere, no subcycling in ACC or FS
+test_name[${index}]="Magnetosphere_small"
+comparison_vlsv[${index}]="bulk.0000001.vlsv bulk.0000002.vlsv"
+comparison_phiprof[${index}]="phiprof_0.txt"
+variable_names[${index}]="proton/vg_rho proton/vg_v proton/vg_v proton/vg_v fg_b fg_b fg_b fg_e fg_e fg_e proton"
+variable_components[${index}]="0 0 1 2 0 1 2 0 1 2"
+((index+=1))
+
+#Very small polar magnetosphere, with subcycling in ACC or FS
+test_name[${index}]="Magnetosphere_polar_small"
+comparison_vlsv[${index}]="bulk.0000001.vlsv bulk.0000002.vlsv"
+comparison_phiprof[${index}]="phiprof_0.txt"
+variable_names[${index}]="proton/vg_rho proton/vg_v proton/vg_v proton/vg_v fg_b fg_b fg_b fg_e fg_e fg_e proton/vg_v_nonthermal proton/vg_v_nonthermal proton/vg_v_nonthermal proton/vg_ptensor_nonthermal_diagonal proton/vg_ptensor_nonthermal_diagonal proton/vg_ptensor_nonthermal_diagonal proton"
+variable_components[${index}]="0 0 1 2 0 1 2 0 1 2 0 1 2 0 1 2"
+((index+=1))
+
+#######
+# GLOBAL 3D TESTS (22..23)
+#######
 
 # Magnetosphere 3D
-test_name[17]="Magnetosphere_3D_small"
-comparison_vlsv[17]="bulk.0000001.vlsv"
-comparison_phiprof[17]="phiprof_0.txt"
-variable_names[17]="proton/vg_rho proton/vg_v proton/vg_v proton/vg_v fg_b fg_b fg_b fg_e fg_e fg_e"
-variable_components[17]="0 0 1 2 0 1 2 0 1 2"
+test_name[${index}]="Magnetosphere_3D_small"
+comparison_vlsv[${index}]="bulk.0000001.vlsv bulk.0000002.vlsv"
+comparison_phiprof[${index}]="phiprof_0.txt"
+variable_names[${index}]="proton/vg_rho proton/vg_v proton/vg_v proton/vg_v fg_b fg_b fg_b fg_e fg_e fg_e"
+variable_components[${index}]="0 0 1 2 0 1 2 0 1 2"
+((index+=1))
 
 # Ionosphere 3D
-test_name[18]="Ionosphere_small"
-comparison_vlsv[18]="bulk.0000001.vlsv"
-comparison_phiprof[18]="phiprof_0.txt"
-variable_names[18]="proton/vg_rho proton/vg_v proton/vg_v proton/vg_v fg_b fg_b fg_b fg_e fg_e fg_e ig_upmappedarea ig_fac ig_rhon ig_potential"
-variable_components[18]="0 0 1 2 0 1 2 0 1 2 0 0 0 0"
-
-# Flowthrough with timevarying inflow
-test_name[19]="Flowthrough_1D_timevarying"
-comparison_vlsv[19]="bulk.0000001.vlsv"
-comparison_phiprof[19]="phiprof_0.txt"
-variable_names[19]="proton/vg_rho proton/vg_v proton/vg_v proton/vg_v fg_b fg_b fg_b fg_e fg_e fg_e"
-variable_components[19]="0 0 1 2 0 1 2 0 1 2 0 0"
-single_cell[19]=1
+test_name[${index}]="Ionosphere_small"
+comparison_vlsv[${index}]="bulk.0000001.vlsv bulk.0000002.vlsv"
+comparison_phiprof[${index}]="phiprof_0.txt"
+variable_names[${index}]="proton/vg_rho proton/vg_v proton/vg_v proton/vg_v fg_b fg_b fg_b fg_e fg_e fg_e ig_upmappedarea ig_fac ig_rhon ig_potential"
+variable_components[${index}]="0 0 1 2 0 1 2 0 1 2 0 0 0 0"
+((index+=1))
+
+
+# choose tests to run (default: all tests)
+run_tests=( )
+for (( c=1; c<$index; c++ ))
+do
+    run_tests+=($c)
+done
+
+# Alternatively, set tests manually, e.g.
+# run_tests=( 1 6 9 )
```

Comment:
not too end-user friendly that these need to be added by numerical index, if these indices are now dynamically generated. So any addition of new tests will now shift the numbers, and requires manual re-numbering.

This could be done by selecting them by their ```test_name```, maybe?
Anyway, it's fine in the context of this PR. Just as soon as we *do* set up smaller tests somewhere (other architectures), this should be reworked.

Reply:
In the past, they were also selected by index, and the index was added manually (and I remember sometimes having issues with tests mixing indexes). So even though that part of it hasn't been drastically improved, it's not actually worse either.

I added currently up-to-date numbers to the test descriptions for now.

Reply:
One day, this will be eg a python script and we can use dictionaries or so?